File: cmh/webapp/src/java/cl/cheekibreeki/cmh/webapp/bl/HorasDisponibles.java
package cl.cheekibreeki.cmh.webapp.bl;

import cl.cheekibreeki.cmh.lib.dal.entities.Bloque;
import java.util.ArrayList;
import java.util.Date;

public class HorasDisponibles {
    private ArrayList<HoraDisponible> horas;
    
    public HorasDisponibles(){
        this.horas = new ArrayList<>();
    }
    
    public HorasDisponibles(Date date, ArrayList<Bloque> bloques){
        this.horas = new ArrayList<>();
        for(Bloque bloque: bloques){
            this.horas.add(new HoraDisponible(date, bloque));
        }
    }
    
    /**
     * @return the horas
     */
    public ArrayList<HoraDisponible> getHoras() {
        return horas;
    }

    /**
     * @param horas the horas to set
     */
    public void setHoras(ArrayList<HoraDisponible> horas) {
        this.horas = horas;
    }
}

File: cmh/terminal/CheekiBreeki.CMH.Terminal.BL/HorasDisponibles.cs
using CheekiBreeki.CMH.Terminal.DAL;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CheekiBreeki.CMH.Terminal.BL
{
    public class HorasDisponibles
    {
        private List<HoraDisponible> horas;

        public HorasDisponibles()
        {
            this.horas = new List<HoraDisponible>();
        }

        public bool bloqueDisponible(BLOQUE bloque)
        {
            foreach (HoraDisponible hora in this.horas)
            {
                if (hora.Bloque.ID_BLOQUE == bloque.ID_BLOQUE)
                {
                    return true;
                }
            }
            return false;
        }

        public HorasDisponibles(DateTime date, List<BLOQUE> bloques)
        {
            this.horas = new List<HoraDisponible>();
            foreach (BLOQUE bloque in bloques)
            {
                this.horas.Add(new HoraDisponible(date, bloque));
            }
        }

        public List<HoraDisponible> getHoras()
        {
            return horas;
        }

        public void setHoras(List<HoraDisponible> horas)
        {
            this.horas = horas;
        }
    }
}
